### PR TITLE
User-authentication

### PR DESCRIPTION
**Pull request by AmerMekic**
# Description

User login with e-mail, without register, any e-mail is valid if it contains the char '@'. User log in information is held in localStorage so we know when the user is logged in. Add to favorites is disabled when the user isn't logged in. Also changed the key, value pairs in localStorage where favorites are held, key is the user email now when the user logs in. Also changed some MUI elements to scale properly with smaller devices.

Task # ([Add authentication](https://app.clickup.com/t/269a11j))

## Screenshots

logout-page

localStorage-for-userinfo-and-favorites

homePage-with-login

login-page

Needs-log-in-to-save-favorites

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Pages/Login.js [hunk 1]
```diff
@@ -1,5 +1,65 @@
...
+  const loggedIn = useSelector((state) => state.userStore.loggedIn);
+  const dispatch = useDispatch();
+  const [email, setEmail] = useState('');
+  const handleClick = () => {
+    //checking if it's an email
+    if (email.includes('@')) {
+      dispatch(addUser(email));
+      window.location.reload();
+    } else {
+      alert('Please enter a valid e-mail');
```
Comment (zlayabekrija) [inline src/Pages/Login.js:19]: Please use toast for this. Let @Dzastinche set up the store for toast first.
https://www.npmjs.com/package/react-toastify

File: src/Pages/Login.js [hunk 2]
```diff
@@ -1,5 +1,65 @@
...
 const Login = () => {
-  return <div>LOGIN</div>;
+  const loggedIn = useSelector((state) => state.userStore.loggedIn);
+  const dispatch = useDispatch();
+  const [email, setEmail] = useState('');
+  const handleClick = () => {
+    //checking if it's an email
+    if (email.includes('@')) {
+      dispatch(addUser(email));
+      window.location.reload();
```
Comment (zlayabekrija) [inline src/Pages/Login.js]: Why do we do reload if we are adding the value to the store?

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Looks great for the first try. Keep up good work. Also please address comments above.

File: src/Pages/Login.js (hunk 2 above)
Comment (AmerMekic) [reply src/Pages/Login.js]: I used reload so it redirects to the homepage after clicking log in

Comment (AmerMekic) [reply src/Pages/Login.js]: Replaced window.location.reload() with useNavigate() so it redirects on login

File: src/Pages/Login.js (hunk 1 above)
Comment (AmerMekic) [reply src/Pages/Login.js:19]: I will wait for him to set up the store

**Review by zlayabekrija — APPROVED** [review]

**Review by RejzovicEmir — APPROVED** [review]

**Pull request merged**